src/dmd/dtoh.d

```diff
@@ -0,0 +1,1586 @@
+    {
+        assert(e.sz == 1 || e.sz == 2);
+        if (e.sz == 2)
+        buf.writeByte('L');
+        buf.writeByte('"');
+        size_t o = buf.offset;
+        for (size_t i = 0; i < e.len; i++)
+        {
+            uint c = e.charAt(i);
+            switch (c)
+            {
+                case '"':
+                case '\\':
+                buf.writeByte('\\');
+                goto default;
+                default:
+                if (c <= 0xFF)
+                {
+                    if (c <= 0x7F && isprint(c))
+                    buf.writeByte(c);
+                    else
+                    buf.printf("\\x%02x", c);
+                }
+                else if (c <= 0xFFFF)
+                    buf.printf("\\x%02x\\x%02x", c & 0xFF, c >> 8);
+                else
+                    buf.printf("\\x%02x\\x%02x\\x%02x\\x%02x", c & 0xFF, (c >> 8) & 0xFF, (c >> 16) & 0xFF, c >> 24);
+                break;
+            }
+        }
+        buf.writeByte('"');
+    }
+    
+    override void visit(AST.RealExp e)
+    {
+        buf.writestring("0");
+    }
+    
+    override void visit(AST.IntegerExp e)
+    {
+        visitInteger(e.toInteger, e.type);
+    }
+    
+    private void visitInteger(dinteger_t v, AST.Type t)
+    {
+        switch (t.ty)
+        {
+            case AST.Tenum:
+                auto te = cast(AST.TypeEnum)t;
+                buf.writestring("(");
+                enumToBuffer(te.sym);
+                buf.writestring(")");
+                visitInteger(v, te.sym.memtype);
+                break;
+            case AST.Tbool:
+                buf.writestring(v ? "true" : "false");
+                break;
+            case AST.Tint8:
+                buf.printf("%d", cast(byte)v);
+                break;
+            case AST.Tuns8:
+            case AST.Tchar:
+                buf.printf("%uu", cast(ubyte)v);
+                break;
+            case AST.Tint16:
+                buf.printf("%d", cast(short)v);
+                break;
+            case AST.Tuns16:
+                buf.printf("%uu", cast(ushort)v);
+                break;
+            case AST.Tint32:
+                buf.printf("%d", cast(int)v);
+                break;
+            case AST.Tuns32:
+                buf.printf("%uu", cast(uint)v);
+                break;
+            case AST.Tint64:
+                buf.printf("%lldLL", v);
+                break;
+            case AST.Tuns64:
+                buf.printf("%lluLLU", v);
+                break;
+            default:
+                //t.print();
+                assert(0);
+        }
+    }
+    
+    override void visit(AST.StructLiteralExp sle)
+    {
+        buf.writestring(sle.sd.ident.toChars());
+        buf.writeByte('(');
+        foreach(i, e; *sle.elements)
+        {
+            if (i)
+            buf.writestring(", ");
+            e.accept(this);
+        }
+        buf.writeByte(')');
+    }
+}
+void genCppFiles(OutBuffer* buf, Modules *ms)
```

Comment:
this and `gencpphdrfiles` should go at the top of the file as they are the only public members.